### PR TITLE
Add Dataset::spatial_ref

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

File: src/dataset.rs [hunk 1]
```diff
@@ -164,6 +164,10 @@ impl Dataset {
         Ok(())
     }
 
+    pub fn spatial_ref(&self) -> Result<SpatialRef> {
+        unsafe { SpatialRef::from_c_obj(gdal_sys::GDALGetSpatialRef(self.c_dataset)) }
+    }
```
Comment (michaelkirk) [inline src/dataset.rs:169]: Would it make sense to add something for GDALSetSpatialRef as well?

Comment (lnicola) [reply src/dataset.rs:169]: I suppose so. And I think the CI will fail for GDAL 2.

**Comment by jdroenner** [comment]
i guess the CI did not fail. So do we merge this now? :D

**Comment by lnicola** [comment]
Wow, I'd forgotten about this PR :smile:.

CI failed, but I fixed it with some `#[cfg]`s.

**Pull request merged**